assembler/src/spades_pipeline/stages/stage.py

```diff
@@ -23,3 +23,20 @@ def generate_config(self, cfg):
 
     def get_command(self, cfg):
         return []
+
+    # shutil.copyfile does not copy any metadata (time and permission), so one
+    # cannot expect preserve_mode = False and preserve_times = True to work.
+    def copy_tree(src, dst, preserve_times = True, preserve_mode = True):
+        if preserve_mode == False:
+            copy_fn = shutil.copyfile
+        else:
+            copy_fn = shutil.copy2
+
+        # shutil.copytree preserves the timestamp, so we must update it afterwards.
+        shutil.copytree(src, dst, copy_function = copy_fn, dirs_exist_ok = True)
```

Comment:
(we support 2.7, 3.6-3.10)
dirs_exist_ok is supported only since 3.8
copy_function is supported only since 3.2

none of this is supported in 2.7